check/src/main.rs

```diff
@@ -32,71 +62,190 @@ struct Args {
     total_messages: u32,
-    warnings: u32,
     skipped: u32,
+}
+
+#[derive(Default)]
+struct LogReport {
+    show_skipped: bool,
+    warnings: u32,
     file_path: String,
 }
 
-impl Report {
-    fn new(file_path: &str) -> Self {
-        Report {
+impl LogReport {
+    fn new(file_path: &str, show_skipped: bool) -> Self {
+        LogReport {
             file_path: file_path.to_string(),
+            show_skipped,
             ..Default::default()
         }
     }
 
-    fn process_row(&mut self, row: AnalysisRow) {
-        self.total_messages += 1;
-        if let Some(reason) = row.skipped_message_reason {
-            *self.skipped_reasons.entry(reason).or_insert(0) += 1;
-            self.skipped += 1;
-            return;
-        }
-        for maybe_event in row.events {
-            let Some(event) = maybe_event else { continue };
-            let Some(timestamp) = row.packet_timestamp else {
-                continue;
-            };
+    fn process_row(&mut self, row: DetectionRow) {
+        for event in row.events {
             match event.event_type {
                 EventType::Informational => {
-                    info!("{}: INFO - {} {}", self.file_path, timestamp, event.message,);
+                    info!(
+                        "{}: INFO - {} {}",
+                        self.file_path, row.packet_timestamp, event.message,
+                    );
                 }
                 EventType::Low | EventType::Medium | EventType::High => {
                     warn!(
                         "{}: WARNING (Severity: {:?}) - {} {}",
-                        self.file_path, event.event_type, timestamp, event.message,
+                        self.file_path, event.event_type, row.packet_timestamp, event.message,
                     );
                     self.warnings += 1;
                 }
             }
         }
     }
 
-    fn print_summary(&self, show_skipped: bool) {
-        if show_skipped && self.skipped > 0 {
+    fn finish(&self, summary: &Summary) {
+        if self.show_skipped && summary.skipped > 0 {
             info!("{}: messages skipped:", self.file_path);
-            for (reason, count) in self.skipped_reasons.iter() {
+            for (reason, count) in summary.skipped_reasons.iter() {
                 info!("    - {count}: \"{reason}\"");
             }
         }
         info!(
             "{}: {} messages analyzed, {} warnings, {} messages skipped",
-            self.file_path, self.total_messages, self.warnings, self.skipped
+            self.file_path, summary.total_messages, self.warnings, summary.skipped
         );
     }
 }
 
-async fn analyze_pcap(pcap_path: &str, show_skipped: bool) {
+struct NdjsonReport {
+    writer: BufWriter<File>,
+}
+
+// The `njson` report has the same output format as the daemon analysis report.
+// See also: [Newline Delimited JSON](https://docs.mulesoft.com/dataweave/latest/dataweave-formats-ndjson)
+impl NdjsonReport {
```

Comment:
This could likely reuse the daemon `AnalysisWriter` struct. Since we're going to be refactoring to reuse components then extracting and reusing that struct will be worth the lift.